frontend/taipy-gui/src/components/Taipy/Status.tsx

```diff
@@ -44,25 +50,104 @@ const status2Color = (status: string): "error" | "info" | "success" | "warning"
     return "info";
 };
 
+// Function to get the appropriate icon based on the status
+const GetStatusIcon = (status: string, withIcons?: boolean): ReactNode => {
+
+    const color = status2Color(status);
+    const iconProps = {
+        sx: { fontSize: 20, color: `${color}.main` },
+    };
+    
+    if (withIcons) {
+        switch (status2Color(status)) {
+            case "success":
+                return <CheckCircleIcon {...iconProps} />;
+            case "warning":
+                return <WarningIcon {...iconProps} />;
+            case "error":
+                return <ErrorIcon {...iconProps} />;
+            default:
+                return <InfoIcon {...iconProps} />;
+        }
+    } else {
+        return getInitials(status);
+    }
+};
+
+
+
 const chipSx = { alignSelf: "flex-start" };
 
 const Status = (props: StatusProps) => {
     const { value, id } = props;
-
+    const content = props.content;
+    const withIcons = props.withIcons;
     const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
 
+    const isSvgUrl = (content?: string) => {
+            return content?.substring(content?.length - 4).toLowerCase() === ".svg"; // Check if it ends with ".svg"
+    };
+
+    const isInlineSvg = (content?: string) => {
+        return content?.substring(0, 4).toLowerCase() === "<svg"; // Check if the content starts with "<svg"
+    };
+
     const chipProps = useMemo(() => {
         const cp: Record<string, unknown> = {};
-        cp.color = status2Color(value.status);
-        cp.avatar = <Avatar sx={{ bgcolor: `${cp.color}.main` }}>{getInitials(value.status)}</Avatar>;
+        const statusColor = status2Color(value.status);
+        cp.color = statusColor;
+       
+        if (isSvgUrl(content)) {
+            cp.avatar = (
+                <Avatar src={content} />
+            );
+        } 
+        
+        else if(content && isInlineSvg(content)){
+            cp.avatar = (
+                <Avatar
+                    sx={{
+                        bgcolor: "transparent"
+                    }}
+                >
+                    <div
+                      style={{
+                        width: '100%',
+                        height: '100%',
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'center',
+                    }}
+                    dangerouslySetInnerHTML={{ __html: content }} />
+                </Avatar>
+            );
+        }
+
+        else {
+            cp.avatar = (
+                <Avatar
+                    sx={{
```

Comment:
useMemo

Reply:
chip.props is already being memoized here.